streaming_data_types/arrays_wa00.py

```diff
@@ -0,0 +1,144 @@
+    x_data_offset = builder.CreateNumpyVector(values_x_array.view(np.uint8))
+    y_data_offset = builder.CreateNumpyVector(values_y_array.view(np.uint8))
+    if y_unit is not None:
+        y_unit_offset = builder.CreateString(y_unit)
+    if x_unit is not None:
+        x_unit_offset = builder.CreateString(x_unit)
+
+    # Build the buffer
+    WaveFormArrayStart(builder)
+    WaveFormArrayAddXData(builder, x_data_offset)
+    WaveFormArrayAddYData(builder, y_data_offset)
+    WaveFormArrayAddXDataType(builder, datatype_x_array)
+    WaveFormArrayAddYDataType(builder, datatype_y_array)
+    if y_timestamp is not None:
+        WaveFormArrayAddYTimestamp(builder, int(y_timestamp.timestamp() * 1e9))
+    if x_timestamp is not None:
+        WaveFormArrayAddXTimestamp(builder, int(x_timestamp.timestamp() * 1e9))
+    if y_unit is not None:
+        WaveFormArrayAddYUnit(builder,y_unit_offset)
+    if x_unit is not None:
+        WaveFormArrayAddXUnit(builder,x_unit_offset)
+    if number_of_elements is not None:
+        WaveFormArrayAddNumberOfElements(builder, number_of_elements)
+
+    WA_Message = WaveFormArrayEnd(builder)
+    builder.Finish(WA_Message, file_identifier=FILE_IDENTIFIER)
+    return bytes(builder.Output())
+
+@dataclass
+class wa00_t:
+    values_x_array: np.ndarray
+    values_y_array: np.ndarray
+    y_timestamp: Optional[np.uint64] = None
+    x_timestamp: Optional[np.uint64] = None
+    x_unit: Optional[str] = None
+    y_unit: Optional[str] = None
+    number_of_elements: Optional[np.uint32] = None
+
+
+
+
+def get_data(raw_data, datatype) -> np.ndarray:
+    """
+    Converts the data array into the correct type.
+    """
+    type_map = {
+        DType.uint8: np.uint8,
+        DType.int8: np.int8,
+        DType.uint16: np.uint16,
+        DType.int16: np.int16,
+        DType.uint32: np.uint32,
+        DType.int32: np.int32,
+        DType.uint64: np.uint64,
+        DType.int64: np.int64,
+        DType.float32: np.float32,
+        DType.float64: np.float64,
+    }
+    return raw_data.view(type_map[datatype])
+
+
+def deserialise_wa00(buffer: Union[bytearray, bytes]) -> wa00_t:
+    check_schema_identifier(buffer, FILE_IDENTIFIER)
+    waveform_array = WaveFormArray.GetRootAsWaveFormArray(buffer, 0)
+    max_time = datetime(
+        year=3001, month=1, day=1, hour=0, minute=0, second=0
+    ).timestamp()
+    y_timestamp = waveform_array.YTimestamp() / 1e9
+    x_timestamp = waveform_array.XTimestamp() / 1e9
+    y_unit = waveform_array.YUnit()
+    x_unit = waveform_array.XUnit()
+
+    if y_timestamp > max_time:
+        y_timestamp = max_time
+    if x_timestamp > max_time:
+        x_timestamp = max_time
+    x_array = get_data(waveform_array.XDataAsNumpy(), waveform_array.XDataType())
+    y_array = get_data(waveform_array.YDataAsNumpy(), waveform_array.YDataType())
+    y_unit = waveform_array.YUnit()
+    x_unit = waveform_array.XUnit()
+    number_of_elements = waveform_array.NumberOfElements()
```

Comment:
You could compare this to the array sizes as a "sanity check", but it's clearly not necessary. I'm wondering whether it was a mistake to suggest that you include it.